packages/kbn-i18n/types/intl_format_cache.d.ts

```diff
@@ -0,0 +1,12 @@
+declare module 'intl-format-cache' {
+  import { IFormats } from 'kbn-i18n-formats';
+
+  interface Message {
+    format: (values: { [key: string]: string | number | Date }) => string;
+  }
+
+  function memoizeIntlConstructor(
+    IntlMessageFormat: any
+  ): (msg: string, locale: string, formats: IFormats) => Message;
```

Comment:
__note:__ it's fine to use `formats: any` here as it's declared inside `intl-message-format`

Reply:
So how we should control `formats` argument type?

Reply:
I'm not worried about controlling type here since it's a temporary type augmentation for the external lib that we'll eventually get rid of and switch to the upstream types (from DefinitelyTyped or from the original repo itself). And we won't have a good way to control that either, so having `formats: unknown` sounds like a good trade off to me.